arrows/pytorch/pytorch_siamese_f_extractor.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import torch
+import torch.utils.data as data
+from torchvision import transforms
+
+from PIL import Image as pilImage
+
+from vital.types import BoundingBox
+from kwiver.arrows.pytorch.models import Siamese
+from kwiver.arrows.pytorch.parse_gpu_list import get_device
+
+
+class siameseDataLoader(data.Dataset):
+    def __init__(self, bbox_list, transform, frame_img, in_size, MOT_flag):
+        self._frame_img = frame_img
+        self._transform = transform
+        self._bbox_list = bbox_list
+        self._mot_flag = MOT_flag
+        self._in_size = in_size
+
+    def __getitem__(self, index):
+        if self._mot_flag is True:
+            bb = self._bbox_list[index]
+        else:
+            bb = self._bbox_list[index].bounding_box()
+
+        im = self._frame_img.crop((float(bb.min_x()), float(bb.min_y()),
+                      float(bb.max_x()), float(bb.max_y())))
+
+        im = im.resize((self._in_size, self._in_size), pilImage.BILINEAR)
+        im.convert('RGB')
+
+        if self._transform is not None:
+            im = self._transform(im)
+
+        return im
+
+    def __len__(self):
+        if self._mot_flag is True:
+            return len(self._bbox_list)
+        else:
+            return self._bbox_list.size()
+
+
+class pytorch_siamese_f_extractor(object):
+    """
+    Obtain the appearance features from a trained pytorch siamese
+    model
+    """
+
+    def __init__(self, siamese_model_path, img_size, batch_size, GPU_list=None):
+        self._device, use_gpu_flag = get_device(GPU_list)
+        # load Siamese model
+        self._siamese_model = Siamese().to(self._device)
+        if use_gpu_flag:
+            self._siamese_model = torch.nn.DataParallel(self._siamese_model, device_ids=GPU_list)
+            snapshot = torch.load(siamese_model_path)
+            self._siamese_model.load_state_dict(snapshot['state_dict'])
+        else:
+            snapshot = torch.load(siamese_model_path, map_location='cpu')
+            tmp = {k[len('module.'):]: v for k, v in snapshot['state_dict'].items()}
```

Comment:
I take it this is meant to strip `"module."` from the start of `k`. Something like this would avoid silently doing the wrong thing:
```python
def strip_prefix(string, prefix):
    if not string.startswith(prefix):
        raise ValueError("{!r} was supposed to start with {!r} but does not".format(string, prefix))
    return string[len(prefix):]
tmp = {strip_prefix(k, 'module.'): v for k, v in snapshot['state_dict'].items()}